Assets/MRTK/Core/Utilities/Editor/AssemblyDefinition.cs

```diff
@@ -223,6 +225,38 @@ public void Save(string fileName)
                 file.IsReadOnly = true;
             }
         }
+
+        /// <summary>
+        /// Adds a reference to an existing assembly definition.  Make sure to call Save() after using this method to save the changes.
+        /// </summary>
+        /// <param name="referenceName">The name of the reference to add to this assembly definition</param>
+        public void AddReference(string referenceName)
+        {
+            if (string.IsNullOrWhiteSpace(referenceName))
+            {
+                Debug.LogError($"The reference name is empty and was not added to the {Name} assembly definition file.");
+                return;
+            }
+
+            if (References == null)
+            {
+                References = new string[] { };
+            }
+
+            List<string> refs = References.ToList();
+
+            if (!refs.Contains(referenceName))
+            {
+                refs.Add(referenceName);
+            }
+            else 
+            {
+                Debug.LogError($"The {referenceName} reference name is already listed in the {Name} file.");
```

Comment:
Do you consider this case an error, or should it just silently return since the reference is present either way?

Reply:
Good catch! This case should not be an error, should just be a log instead of a log error

Reply:
Added